fdbserver/include/fdbserver/StorageMetrics.actor.h

```diff
@@ -152,3 +157,76 @@ struct ByteSampleInfo {
 // Determines whether a key-value pair should be included in a byte sample
 // Also returns size information about the sample
 ByteSampleInfo isKeyValueInSample(KeyValueRef keyValue);
+
+class IStorageMetricsService {
+public:
+	StorageServerMetrics metrics;
+
+	// penalty used by loadBalance() to balance requests among service instances
+	virtual double getPenalty() { return 1; }
+
+	virtual bool isReadable(KeyRangeRef const& keys) { return true; }
+
+	virtual void addActor(Future<Void> future) = 0;
+
+	virtual void getSplitPoints(SplitRangeRequest const& req) = 0;
+
+	virtual Future<Void> waitMetricsTenantAware(const WaitMetricsRequest& req) = 0;
+
+	virtual void getStorageMetrics(const GetStorageMetricsRequest& req) = 0;
+
+	// NOTE: also need to have this function but template can't be a virtual so...
+	// template <class Reply>
+	// void sendErrorWithPenalty(const ReplyPromise<Reply>& promise, const Error& err, double penalty);
+};
+
+ACTOR template <class ServiceType>
+Future<Void> serveStorageMetricsRequests(ServiceType* self, StorageServerInterface ssi) {
```

Comment:
It looks like the `IStorageMetricsService` interface is never actually used for dynamic polymorphism. Why not add a method:
```
virtual Future<Void> serveStorageMetricsRequests(StorageServerInterface const &ssi)=0;
```
This would allow the implementation to be moved out of the header file, and this file could be changed back to a `.h` file.